setzer/document/preview/preview_controller.py

```diff
@@ -166,6 +179,16 @@ def on_button_press(self, widget, event):
 
             return True
 
+    def on_pan(self, widget, event):
+        if event.state & Gdk.ModifierType.BUTTON2_MASK:
+            sw_hadj = self.view.scrolled_window.get_hadjustment()
+            sw_vadj = self.view.scrolled_window.get_vadjustment()
+            sw_hadj.set_value(sw_hadj.get_value() + self.click_x - event.x)
+            sw_vadj.set_value(sw_vadj.get_value() + self.click_y - event.y)
+
+    def on_button_release(self, widget, event):
+        self.view.scrolled_window.get_window().set_cursor(self.cursor_default)
```

Comment:
It isn't really required, since `update_cursor` changes it when moving the cursor but at some point I've noticed it during testing.